kernel/src/log_segment.rs

```diff
@@ -162,4 +396,81 @@ mod tests {
         // missing columns, but can still skip part 3 because has valid nullcount stats for P&M.
         assert_eq!(data.len(), 4);
     }
+
+    #[test]
```

Comment:
can we add a few tests:
1. fail for logsegment that's not continuous
2. check on a CDF-style log segment (build one with omit_checkpoints = true) that has checkpoints in the range
3. sorting tests for commits and checkpoints
4. check for LogSegment created with checkpoint hint (to check that bug above where I switched `>=` to `<= cp.version`

Reply:
Yeah, a few more tests would be good. Also:
- Errors if start_version and start_checkpoint are both specified